### ISSUE TITLE
2 ether bounty - How should share price be determined

### ISSUE THREAD

**Issue by peerther** [peerther/Smart-Contract-based-Organization#2]
2 ether bounty:
Best 2 ideas each win the bounty.
Within this contract, what would be the way to hold the share price. In the beginning, a founder usually will set a 1 penny per share price, but as investment comes in, the shares value changes based on how much the investor paid per share. 
Additionally, the revenue for the company could go up, and the share value could increase without investors. 
Debt could come into the company as capital, and increase the share capital due to new assets the company acquires with that debt.
Given that, how should we handle the setting of share price, other than a 50% of more vote on what the price should be by common share holders?

**Comment by sinking-point** [comment]
What about auctioning them off?

**Comment by peerther** [comment]
What would that look like from an implementation perspective?

**Comment by D-Nice** [comment]
Isn't it best for it to be based on the total amount of ether deposited, within the company contract, + a premium percentage. That would make the terminateCompany() function more valid, and also encourage investment within the company. Amounts sent to the company treasury would not lower this deposit amount, as those should be considered funds used by the company to increase its value, although there would have to be a "fractional-reserve" system in place, where a good percentage of funds have to stay within the company contract, in case it goes bust, so that investors can get at least something back. 

It would be fair for the share premium in fact to be equal to the fractional-reserve percentage. Hence if you have to pay a 50% premium on share price, you pay more for less risk, as the company would have to keep at least 50% of all investments within the company contract.

Having a honours bachelor in commerce, I think this would be the "fairest" system and closest to free-market, when considering the company will not be on a publicly-traded exchange. Another feature implemented should be the ability of the company to buy back shares from certain investors, and a "CEO" modifier should be assigned to the original creator, however, as in a real company, shareholder majority should be able to override that position.

**Comment by peerther** [comment]
Yeah, good point. I think there should be a distinction made. As is, the shares are issued to people and shares can be "traded". Important distinction. If shares are given to you by a majority vote from shareholders, than those shares each costed the default .0001 ether at the creation of the company. However, the value isn't warchest for the company. The warchest is simply where ether lives that is sent to the contract. If someone buys ether from someone else, then the individual who sold them should have the value, and it should not be in the warchest. If a VC gives cash to a company, the company issues shares to that investor on this network, but the money transaction didn't happen here, it's just being represented here. The "signing" of a contract with the vc, would be sending them x shares, and specifying a new share price that's agreed upon. The only problem is, immediately after the deal, the share price is ether more or less valuable. 
So, given that there is a representation of value for when money is transacted offchain, as well as, ether transactions whereby user sell their shares to other parties (along with the rights if they are common), maybe yeah, D-Nice, the last traded transaction for when shares are traded instead of issued should set a "price", but this is arbitrary... and unless there is a "public pool" (where company has a buy-back price, and users can also sell /buy to the contract address without dealing directly with people)... which would be cool, actually, there isn't a public market, but yet it's still private... just the world can see it on the blockchain.
D-Nice, what do you think?

**Comment by peerther** [comment]
Public note from Peerther to D-Nice: I will leave it up to your discretion to make a call here. But the important thing would be the difference between when shares are traded on the ether network directly, versus being a representation of something that happened off-chain with, say, a VC. The treasure chest (warchest), is just where the ether that it costs for each share to be "created" lives. If I own a share and sell it to you, I should get the value, not the corporate treasure chest. If shareholders issue shares (as a new class) from nowhere or transfer from ESOP to an individual from a vote, the shareholders will declare a price (int amount and string currency). Later when we add a "automated buy-back program and a public pool" then the market will be setting the price. I'll create a bounty now for that additional feature.

**Comment by peerther** [comment]
D-Nice, what ether address should we use to issue this bounty. I will also keep this open for new ideas, and any good ones that are better will get 2 ether.

**Comment by D-Nice** [comment]
Thanks, you can send it to 0x13c7c27ea7ee89ec6607965b6f357b0109bc98b4

Also, as I am implementing the sell shares function, another possibility would be simply to use the last successful sale price, just like a exchange would, or to avoid erratic prices, or people purposely selling cents on the dollar, would be a weighted-average system. However, it depends what price we are talking about, if we are talking about IPO price, that is completely handled by the company, and what worth it thinks it has (usually decided by professional accounting/investment firm in real life). However, the value of the company's stock should be determined by market forces thereafter.

**Comment by peerther** [comment]
Ether sent: https://etherscan.io/tx/0xc4ac09493e806908cc7e925ad372d97a831e4a7ca2362cf1163953effa301719
I agree with using last price, but let's say someone gives me cash outside of ethereum, yet I transfer them all my shares for free in ethereum... that would mean share price would go to 0. So, I guess shareholders can vote for the price if there isn't an "IPO". Soon, we'll implement the "IPO" and the company will set the price and amount they will "buy back", and people can just send funds to the contract and get shares, and also list their shares for sell for a certain price. Then we have a real market, and that market can set the price, because people are transacting with ether. Before that, ether is just a represnetation of other currencies outside of ethereum (acting as a ledger). Hope that makes sense...

**Comment by D-Nice** [comment]
Is there any reason someone would prefer to buy outside of the contract, rather than securely buy through the contract itself? Right now I'm basically writing a built-in "exchange" system within the contract.

**Comment by peerther** [comment]
For sure. most VCs and Angels will still be writing checks to blockchain based companies that use this contract, and the goal is to have their investments represented on the blockchain. This is default. In future we will have a: runIPO function(priceOffered, priceWillByBackAfterOfferFilled, amountOffered, amountWillAutobuyBack). So, we should be able to support both... that's good.
